modules/Bio/EnsEMBL/DBSQL/ArchiveStableIdAdaptor.pm

```diff
@@ -383,16 +399,21 @@ sub _fetch_archive_id {
   my $extra_sql1 = shift;
   my $extra_sql2 = shift;
 
+# Retrieve species_id value from parent "adaptor" class
+  my $species_id = $self->species_id();
+
   # using a UNION is much faster in this query than somthing like
   # "... AND (sie.old_stable_id = ? OR sie.new_stable_id = ?)"
   my $sql = qq(
     SELECT * FROM stable_id_event sie, mapping_session ms
     WHERE sie.mapping_session_id = ms.mapping_session_id
+    AND ms.species_id = $species_id
```

Comment:
We try not to interpolate variables directly into SQL, not least of which to avoid injection attacks. They should be "?" and in the execute() parameters